tests/iio_attr.c

```diff
@@ -528,21 +663,33 @@ int main(int argc, char **argv)
 		return EXIT_FAILURE;
 	if (wbuf && !wbuf)
 		return EXIT_FAILURE;
-	if (wbuf && ((device_index && (!strcmp(".", argv[device_index]) ||
+	if ((gen_code || wbuf) && ((device_index && (!strcmp(".", argv[device_index]) ||
 				        strchr(argv[device_index], '*'))) ||
 		     (channel_index && (!strcmp(".", argv[channel_index]) ||
 					 strchr(argv[channel_index], '*'))) ||
 		     (attr_index && (!strcmp(".", argv[attr_index])  ||
 				      strchr(argv[attr_index], '*'))))) {
-		printf("can't write value with wildcard match\n");
+		printf("can't %s with wildcard match\n",
+				gen_code ? "generate code" : "write value");
 		return EXIT_FAILURE;
 	}
 
+
+	if (gen_code) {
+		gen_fd = fopen (gen_file, "w");
+		if (!gen_fd) {
+			errnum = errno;
+			printf("Error '%s' opening file: %s\n", strerror(errnum),gen_file);
+			return EXIT_FAILURE;
+		}
+	}
+
 	if (detect_context)
-		ctx = autodetect_context();
-	else if (backend == AUTO)
+		ctx = autodetect_context(gen_code, gen_fd);
+	else if (backend == AUTO) {
 		ctx = iio_create_context_from_uri(arg_uri);
-	else
+		gen_start_uri(arg_uri, gen_fd);
```

Comment:
is there a point where the `(backend == AUTO)` && `gen_code` are synchronized?
would we need to make this 
```
if (gen_code)
  gen_start_uri(arg_uri, gen_fd);
```
?

Reply:
when autoscanning for a context (over USB), the uri is never passed back, and isn't part of the actual context - so to keep things simple, the call togen_start_uri(arg_uri, gen_fd);  is added inside the autodetect_context function.

Reply:
ack